manager/controlapi/secret.go

```diff
@@ -12,13 +18,41 @@ import (
 // MaxSecretSize is the maximum byte length of the `Secret.Spec.Data` field.
 const MaxSecretSize = 500 * 1024 // 500KB
 
+var isValidSecretName = regexp.MustCompile(`^[a-zA-Z0-9](?:[a-zA-Z0-9-_.]{0,62}[a-zA-Z0-9])*$`)
+
+// assumes spec is not nil
+func secretFromSecretSpec(spec *api.SecretSpec) *api.Secret {
+	checksumBytes := sha256.Sum256(spec.Data)
+	return &api.Secret{
+		ID:         identity.NewID(),
+		Spec:       *spec,
+		SecretSize: uint32(len(spec.Data)),
+		Digest:     hex.EncodeToString(checksumBytes[:]),
```

Comment:
Ah, thanks for the link to that package!  I originally had the algorithm in the front, but it made the display of the hash a little longer, so removed it, but agree it's a better idea to include.  And thanks for the quick review!